subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -43,25 +44,65 @@ static int tp_udp_init(struct mqtt_sn_transport *transport)
 {
 	struct mqtt_sn_transport_udp *udp = UDP_TRANSPORT(transport);
 	int err;
+	struct sockaddr_in addrm;
```

Comment:
So no IPv6 support?

Reply:
Not currently in the provided implementation but that matches the current implementation that I am extending. This does not preclude the user from implementing their own transport that uses IPv6. Should I add support as part of this PR or leave the current implementation's limitation?

Reply:
As the other parts of the Zephyr network stack typically support IPv6 and IPv4 (dual stack), it would be preferable if all new code also support dual stack. So yes, I would prefer IPv6 support also in this PR.